src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -318,4 +318,13 @@ object messages {
         |$code2
         |""".stripMargin
   }
+
+  case class IllegalExpressionStart()(implicit ctx: Context) extends Message(14) {
+    val kind = "Syntax"
+    val msg = "illegal start of simple expression"
+    val explanation = {
+      hl"""|The expression you've provided is not valid because
+           |an expression in Scala always returns a value of some type""".stripMargin
```

Comment:
If the explanation is printed, the user is most likely a beginner. In this case we need to educate them by providing a hint as to how they can make their program valid. If I don't know that expressions are things which return values then I, the user, need to be given some examples.

This error also occurs frequently when a user has forgotten a parenthesis somewhere which makes it sort of tricky to  give a 100% concrete solution in this case. Perhaps in the future we could have some machinery to discover this.

I would try to write something similar to:

```
An expression yields a value. In the case of the simple expression, this error
commonly occurs when there's a missing parenthesis or brace. The reason being
that a simple expression is one of the following:

- Block
- Expression in parenthesis
- Identifier
- Object creation
- Literal

Which cannot start with a `}` or a `)`.
```